pkg/login/social/social.go

```diff
@@ -85,6 +85,13 @@ func NewOAuthService() {
 
 	for _, name := range allOauthes {
 		sec := setting.Raw.Section("auth." + name)
+
+		// when empty_scopes parameter exists and is true, overwrite scope with empty value
+		if sec.HasKey("empty_scopes") {
+			if sec.Key("empty_scopes").Value() == "true" {
+				sec.Key("scopes").SetValue("")
+			}
+		}
```

Comment:
I would suggest doing the check after `info` has been initialized below rather than modifying the section map. Something along these lines

```go
if sec.Key("empty_scopes").MustBool() {
  info.Scopes = []string{}
}